pep-0554.rst

```diff
@@ -282,107 +241,109 @@ Synchronize using a channel
 ::
 
    interp = interpreters.create()
-   r, s = interpreters.create_channel()
-   def run():
-       interp.run(tw.dedent("""
-           reader.recv()
+   r, s = os.pipe()
+   print('before')
+   interp.run(tw.dedent("""
+           import os
+           os.read(reader, 1)
            print("during")
            """),
            shared=dict(
                reader=r,
                ),
            )
-   t = threading.Thread(target=run)
-   print('before')
-   t.start()
    print('after')
-   s.send(b'')
+   os.write(s, '')
 
 Sharing a file descriptor
 -------------------------
 
 ::
 
    interp = interpreters.create()
-   r1, s1 = interpreters.create_channel()
-   r2, s2 = interpreters.create_channel()
-   def run():
-       interp.run(tw.dedent("""
+   r1, s1 = os.pipe()
+   r2, s2 = os.pipe()
+   interp.run(tw.dedent("""
+           import os
            fd = int.from_bytes(
-                   reader.recv(), 'big')
+                   os.read(reader, 10), 'big')
            for line in os.fdopen(fd):
                print(line)
-           writer.send(b'')
+           os.write(writer, b'')
            """),
            shared=dict(
-               reader=r,
+               reader=r1,
                writer=s2,
                ),
            )
-   t = threading.Thread(target=run)
-   t.start()
    with open('spamspamspam') as infile:
        fd = infile.fileno().to_bytes(1, 'big')
-       s.send(fd)
-       r.recv()
+       os.write(s1, fd)
+       os.read(r2, 1)
 
 Passing objects via marshal
 ---------------------------
 
 ::
 
    interp = interpreters.create()
-   r, s = interpreters.create_channel()
+   r, s = os.pipe()
    interp.run(tw.dedent("""
        import marshal
+       import os
        """),
        shared=dict(
            reader=r,
            ),
-       )
-   def run():
-       interp.run(tw.dedent("""
-           data = reader.recv()
-           while data:
+       ).wait()
+   interp.run(tw.dedent("""
+           data = b''
+           c = os.read(reader, 1)
+           while c != b'\x00':
+               while c != b'\x00':
+                   data += c
+                   c = os.read(reader, 1)
                obj = marshal.loads(data)
                do_something(obj)
-               data = reader.recv()
+               c = os.read(reader, 1)
            """))
-   t = threading.Thread(target=run)
-   t.start()
    for obj in input:
        data = marshal.dumps(obj)
-       s.send(data)
-   s.send(None)
+       os.write(s, data)
+       os.write(s, b'\x00')
+   os.write(s, b'\x00')
```

Comment:
The `marshal` example doesn't provide anything new over the `pickle` one. Perhaps just add a note in `pickle` that `marshal` can be used exactly the same way?
(I'm preferring `pickle` because the [marshal docs](https://docs.python.org/3/library/marshal.html) recommend it for this kind of use.)

Reply:
I've dropped the "marshal" example.